packages/web-components/fast-router/src/links.ts

```diff
@@ -0,0 +1,88 @@
+import { Navigation } from "./navigation";
+
+interface AnchorEventInfo {
+    shouldHandleEvent: boolean;
+    href: string | null;
+    anchor: Element | null;
+}
+
+export interface LinkHandler {
+    connect(): void;
+    disconnect(): void;
+}
+
+export class DefaultLinkHandler implements LinkHandler {
+    private handler = (e: MouseEvent) => {
+        let { shouldHandleEvent, href } = this.getEventInfo(e);
+
+        if (shouldHandleEvent) {
+            e.preventDefault();
+            Navigation.push(href!);
+        }
+    };
+
+    public connect(): void {
+        window.addEventListener("click", this.handler, true);
+    }
+
+    public disconnect(): void {
+        window.removeEventListener("click", this.handler);
+    }
+
+    private getEventInfo(event: MouseEvent): AnchorEventInfo {
+        const info: AnchorEventInfo = {
+            shouldHandleEvent: false,
+            href: null,
+            anchor: null,
+        };
+
+        const target = this.findClosestAnchor(event);
+
+        if (!target || !this.targetIsThisWindow(target)) {
+            return info;
+        }
+
+        if (
+            target.hasAttribute("download") ||
+            target.hasAttribute("router-ignore") ||
+            target.hasAttribute("data-router-ignore")
+        ) {
+            return info;
+        }
+
+        if (event.altKey || event.ctrlKey || event.metaKey || event.shiftKey) {
+            return info;
+        }
+
+        const href = target.getAttribute("href");
```

Comment:
Maybe we should add a check for href above to prevent my previous comment about this potentially being a button submit?